src/lib/Lib/AbsoluteCompilationDatabase.cpp

```diff
@@ -51,14 +51,22 @@ static
 std::vector<std::string>
 adjustCommandLine(
     const std::vector<std::string>& cmdline,
-    const std::vector<std::string>& additional_defines)
+    const std::vector<std::string>& additional_defines,
+    std::unordered_map<std::string, std::vector<std::string>> const& includePathsByCompiler)
 {
     std::vector<std::string> new_cmdline;
     std::vector<std::string> discarded_cmdline;
     llvm::opt::InputArgList args;
     StringRef driver_mode;
-    if(! cmdline.empty())
+
+    std::vector<std::string> systemIncludePaths;
+
+    if( ! cmdline.empty())
     {
+        if (auto it = includePathsByCompiler.find(cmdline[0]); it != includePathsByCompiler.end()) {
+            systemIncludePaths = it->second;
```

Comment:
There's no need to copy it because we're not mutating it. A reference or a pointer (depending on how it's implemented) should be enough here.